core/01_core.js

```diff
@@ -161,15 +115,10 @@
 
   function opAsync(opName, ...args) {
     const promiseId = nextPromiseId++;
-    let p = setPromise(promiseId);
-    try {
-      ops[opName](promiseId, ...args);
-    } catch (err) {
-      // Cleanup the just-created promise
-      getPromise(promiseId);
-      // Rethrow the error
-      throw err;
-    }
+    let p = newPromise();
+    // Save the id on the promise so it can later be ref'ed or unref'ed
+    p[promiseIdSymbol] = promiseId;
```

Comment:
The promise ID symbol has to be set on the promise returned by `opAsync`, not on the intermediate promise, since otherwise the ID is not available to extensions that need to use `{un,}refOp`.